### PR TITLE
Make docker images multiplatform

### PR DESCRIPTION
**Pull request by Hoverbear**
Follows from the lessons in https://docs.docker.com/build/building/multi-platform/ and incorporates some of #595 .

This may fail in CI as it does require a BuildKit compatible docker and I have no idea what CircleCI uses.

### CODE REVIEW

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
Holy 🐮 , it just worked???

File: ferrocene/ci/docker-images/ubuntu-20/Dockerfile [hunk 1]
```diff
@@ -91,34 +118,56 @@ RUN mkdir /tmp/python \
...
-RUN curl -Lo /tmp/awscli.zip https://awscli.amazonaws.com/awscli-exe-linux-x86_64-2.12.4.zip \
-    && echo "0556162145e68889a27f835307b9dad56de60fda4256d2e246e9699f24535a08  /tmp/awscli.zip" | sha256sum -c \
-    && unzip -q -d /tmp/awscli /tmp/awscli.zip \
-    && /tmp/awscli/aws/install \
-    && rm -rf /tmp/awscli /tmp/awscli.zip
+# Install AWS CLI from the published binaries, as no source tarball is provided
+# and the version in the Ubuntu 24.04 is a snap.
+RUN <<-EOF
+    set -xe
+    URL=""
```
Comment (tshepang) [inline ferrocene/ci/docker-images/ubuntu-20/Dockerfile:138]: looks unused
```suggestion
```

File: ferrocene/ci/docker-images/ubuntu-24/Dockerfile [hunk 2]
```diff
@@ -27,41 +35,70 @@ RUN apt-get update \
...
 # Install AWS CLI from the published binaries, as no source tarball is provided
 # and the version in the Ubuntu 24.04 is a snap.
-RUN curl -Lo /tmp/awscli.zip https://awscli.amazonaws.com/awscli-exe-linux-x86_64-2.12.4.zip \
-    && echo "0556162145e68889a27f835307b9dad56de60fda4256d2e246e9699f24535a08  /tmp/awscli.zip" | sha256sum -c \
-    && unzip -q -d /tmp/awscli /tmp/awscli.zip \
-    && /tmp/awscli/aws/install \
-    && rm -rf /tmp/awscli /tmp/awscli.zip
+RUN <<-EOF
+    set -xe
+    URL=""
```
Comment (tshepang) [inline ferrocene/ci/docker-images/ubuntu-24/Dockerfile:60]: looks unused also
```suggestion
```

**Review by tshepang — APPROVED** [review]

**Review by tshepang — APPROVED** [review]
bors merge

File: ferrocene/ci/docker-images/ubuntu-20/Dockerfile [hunk 3]
```diff
@@ -91,34 +118,54 @@ RUN mkdir /tmp/python \
...
 # you will unfortunately need to change this to build NodeJS from source.
-RUN mkdir /opt/nodejs \
-    && curl -Lo /tmp/nodejs.tar.xz https://nodejs.org/dist/v16.20.2/node-v16.20.2-linux-x64.tar.xz \
-    && echo "874463523f26ed528634580247f403d200ba17a31adf2de98a7b124c6eb33d87 /tmp/nodejs.tar.xz" | sha256sum -c \
-    && tar xf /tmp/nodejs.tar.xz -C /opt/nodejs --strip-components=1 \
-    && rm /tmp/nodejs.tar.xz \
-    && /opt/nodejs/bin/node --version
+RUN <<-EOF
+    mkdir /opt/nodejs
+    curl -Lo /tmp/nodejs.tar.xz https://nodejs.org/dist/v16.20.2/node-v16.20.2-linux-x64.tar.xz
```
Comment (pietroalbini) [inline ferrocene/ci/docker-images/ubuntu-20/Dockerfile:123]: We are installing the prebuilt nodejs for just x86-64 here.

**Pull request merged**